Controller/FragmentAdminController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ArticleBundle\Controller;
+
+use Sonata\AdminBundle\Controller\CRUDController;
+use Symfony\Component\Form\Form;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+/**
+ * @author Hugo Briand <briand@ekino.com>
+ */
+class FragmentAdminController extends CRUDController
+{
+    /**
+     * @return Response
+     *
+     * @throws \RuntimeException
+     * @throws AccessDeniedException
+     */
+    public function formAction()
+    {
+        if (false === $this->admin->isGranted('CREATE')) {
+            throw new AccessDeniedException();
+        }
+
+        $object = $this->admin->getNewInstance();
+
+        $this->admin->setSubject($object);
+
+        /** @var $form Form */
+        $form = $this->admin->getForm();
+        $form->remove('type')->add('type', 'hidden'); // Hide type field when adding
+        $form->setData($object);
+
+        $view = $form->createView();
+
+        // Set the theme for the current Admin Form
+        $this->get('twig')->getExtension('form')->renderer->setTheme($view, $this->admin->getFormTheme());
+
+        // We need to replace name attributes to have a s20003903[fragments][n][...] format as this form doesn't
+        // Know of its parent: s/s20003903_fragments_n/s20003903[fragments][n]
+        $search = sprintf('name="%s_%d', $this->get('request')->get('elementId'), $this->get('request')->get('fragCount', 0));
+        $replace = sprintf('name="%s[fragments][%d]', $this->get('request')->get('uniqid'), $this->get('request')->get('fragCount', 0));
```

Comment:
Done. Called getRequest method that provide currentRequest from request stack service